### DISCUSSION TITLE
Typescript-eslint should use its own "recommended" config and have a simple startup example

### DISCUSSION THREAD

**Discussion by Tanavya** [typescript-eslint/typescript-eslint#8951] [Community Feedback]
I was looking at this file here https://github.com/typescript-eslint/typescript-eslint/blob/main/eslint.config.mjs.

This file is so big adds so many custom rules instead of just the "recommended" from `tseslint.configs.recommended`, even more weirdly it uses non-conventional file extension `eslint.config.mjs` with a TODO in the `eslint.config.mjs`. 

Idk if I am too inexperienced in all the intricacies of typescript, but I am not sure why the configuration of typescript-eslint has to be so complicated. I think it would help if there was a simple example application managed by typescript-eslint, with eslint and prettier and other basic, "recommended" configuration already set up.

**Comment by bradzacher** [comment]
> weirdly it uses non-conventional file extension eslint.config.mjs

This is not non-conventional - it's officially supported as of ESLint v8.57.0.
https://eslint.org/blog/2024/02/eslint-v8.57.0-released/#support-for-eslint.config.mjs-and-eslint.config.cjs
It's also required if we want to use ESM in the config file because our project has a hard requirement on being CJS.

The reason we have the `eslint.config.js` left behind is due to our dependency on `nx` and it not supporting `.mjs` files yet.

**Comment by bradzacher** [comment]
> I think it would help if there was a simple example application managed by typescript-eslint, with eslint and prettier and other basic, "recommended" configuration already set up.

We have https://github.com/typescript-eslint/typescript-eslint-examples
It's just not something we've spent much time on.

Have you checked out our getting started guide?
It should help you in bootstrapping your local setup:
https://typescript-eslint.io/getting-started/

**Comment by bradzacher** [comment]
> why the configuration of typescript-eslint has to be so complicated

You mean "the configuration of eslint".
We are a 3rd party within the ESLint ecosystem and are beholden to the standards and tools built by ESLint itself.

**Comment by bradzacher** [comment]
> instead of just the "recommended" from tseslint.configs.recommended

We do use the recommend, but more.
https://github.com/typescript-eslint/typescript-eslint/blob/main/eslint.config.mjs#L67-L68

As per our config docs
https://typescript-eslint.io/users/configs#recommended-configurations

the strict config is the "recommended" and more.

**Comment by bradzacher** [comment]
> This file is so big adds so many custom rules

That's part of ESLint really - for any non-trivial usecase it's not going to be a simple "drop a config and go" thing. It's going to require effort to handle edgecases where you need to turn things on/off (eg test files have different constraints to non-test files).
You're also gonna wanna turn on additional rules above recommended configs or off rules from recommended configs.
You're also gonna turn on plugins that might not have recommended configs.

Configuration is just a messy business and codebases aren't a homogeneous blob of code.

**Comment by bradzacher** [comment]
Finally, as per our contributing guide: https://typescript-eslint.io/contributing/discussions
We don't use discussions for this purpose. See the guide for links to alternatives such as stackoverflow or discord.
